server/integration/feature-pack/src/main/resources/modules/system/layers/base/org/hibernate/search/engine/main/module.xml

```diff
@@ -8,6 +8,7 @@
         <module name="javax.transaction.api" />
         <module name="org.hibernate.commons-annotations" />
         <module name="org.apache.lucene" export="true" />
+        <module name="org.apache.lucene.internal" export="true" />
```

Comment:
It seems exporting "org.apache.lucene.internal" was not need. I've added it as a regular dependency and it's enough to fix the problem. 

No need to do anything for the wildfly-modules; they work as they are because the hibernate search modules do not have this issue. 

I think it should be all right now.